services/ui-src/src/utils/other/parsing.ts

```diff
@@ -38,12 +49,34 @@ export const parseCustomHtml = (element: CustomHtmlElement[] | string) => {
         content = sanitizeAndParseHtml(content);
         // delete 'as' prop since React.Fragment can't accept it
         delete elementProps.as;
+        return React.createElement(elementType, elementProps, content);
       }
-      return React.createElement(elementType, elementProps, content);
+      return createElementWithChildren(element);
     });
   }
 };
 
+export function createElementWithChildren(
+  element: CustomHtmlElement
+): ReactElement {
+  const { type, content, as, props } = element;
+  const elementType: string = customElementMap[type] || type;
+  const elementProps = {
+    key: type + `${Math.floor(Math.random() * 100)}`,
```

Comment:
Assume we are doing this because we need a unique key? What's the effect of we have a duplicate?

Reply:
> Assume we are doing this because we need a unique key? What's the effect of we have a duplicate?

Yeah, previously this was just the index of the field in the list of all fields. Duplicate would cause an error at least in the console, not sure if a breaking one.

Reply:
@tblackwe I changed this so we now generate a UUID for the repeated elements